tsdb/engine.go

```diff
@@ -60,6 +74,21 @@ func NewEngine(path string, walPath string, options EngineOptions) (Engine, erro
 	// Only bolt-based backends are currently supported so open it and check the format.
 	var format string
 	if err := func() error {
+		// if it's a dir then it's a tsm1 engine
```

Comment:
And what would happen if this code did open a later engine without checking for the magic string? And then writes data into it? Best case it might error and do nothing. Worse case it might corrupt the engine it's writing into. A newer engine may be different in only subtle ways, not different enough to cause errors, but perhaps different enough to corrupt existing data.